File: duplicates.java
// TC: O(n)
// SC: O(1)

class Solution {
    public int removeDuplicates(int[] nums) {
        int s = 0;
        int count = 1;

        for(int f = 1; f< nums.length; f++){
            if(nums[f] == nums[f-1]){
                count++;
            }
            else{
                count = 1;
            }
            if(count <= 2){
                s++;
                nums[s] = nums[f];
            }
        }
        return s+1;
    }
}

File: Duplicates.cs
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : yes

public class Solution {
    public int RemoveDuplicates(int[] nums) {
        if(nums == null ||  nums.Length == 0) return 0;
        int slow = 1;        
        int count = 1;
        for(int i = 1; i< nums.Length; i++)
        {
            if(nums[i] == nums[i-1] )
            {               
                count++;
            }
            else{
                count =1;
            }
            
            if(count <= 2)
            {
                nums[slow] = nums[i];
                slow++;
                
            }  
                                 
        }
        return slow;
    }
}
